### PR TITLE
Feature/folder sync

### PR DESCRIPTION
**Pull request by GitBartlomiej**
First pull req

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -15,29 +18,70 @@ namespace fs = std::filesystem;
...
 int main() {
     std::cout
             << "Hello world ... bo od czegoś trzeba zacząć ... za kilka dni będzie to potężny program do synchronizowania plików xD xD xD\n";
 
+    std::cout << "Source: " << source << std::endl;
+    std::cout << "Destination: " << destination << std::endl;
+    std::cout << currentPath << std::endl;
+
+    std::mutex mutex;
+    std::condition_variable cv;
```
Comment (mkijew) [inline main.cpp:76]: cv - probably is not necessary

Comment (GitBartlomiej) [reply main.cpp:76]: ok

File: main.cpp [hunk 2]
```diff
@@ -7,6 +7,9 @@
 #include <filesystem>
 
 namespace fs = std::filesystem;
+fs::file_time_type sourceTimePrevious;
+
+bool once = true;
```
Comment (mkijew) [inline main.cpp]: consider changing it atomic
at present std::unique_lock is fair enough :)

File: main.cpp [hunk 3]
```diff
@@ -15,29 +18,70 @@ namespace fs = std::filesystem;
...
+    cv.wait(lock, [&isReady] { return isReady; });
+
+    //check once
+    if (once) {
+        sourceTimePrevious = fs::last_write_time(source);
+        once = false;
+    }
+    auto sourceTime = fs::last_write_time(source);
 
+    //Check recursively if the source and destination folders have the same number of subfolders
```
Comment (mkijew) [inline main.cpp]: correct comment

File: main.cpp [hunk 4]
```diff
@@ -15,29 +18,70 @@ namespace fs = std::filesystem;
...
+                if (sourceTime != sourceTimePrevious) {
+                    std::cout << "Source folder has been modified" << std::endl;
+                    std::cout << "Source time: " << sourceTime.time_since_epoch().count() << std::endl;
+                    std::cout << "sourceTimePrevious time: " << sourceTimePrevious.time_since_epoch().count()
+                              << std::endl;
 
-    for (const auto &entry: fs::directory_iterator(source)) {
-        if (fs::is_directory(entry)) {
-        } else {
+                    fs::remove_all(destinationPath);
```
Comment (mkijew) [inline main.cpp]: For the first time it is OK. But this code should be improved.

File: main.cpp [hunk 5]
```diff
@@ -15,29 +18,70 @@ namespace fs = std::filesystem;
...
 
+    std::cout << "Source: " << source << std::endl;
+    std::cout << "Destination: " << destination << std::endl;
+    std::cout << currentPath << std::endl;
+
+    std::mutex mutex;
+    std::condition_variable cv;
+
+    do {
+        addDiffrenceFileToTemp(source, destination, mutex, cv, true);
```
Comment (mkijew) [inline main.cpp:80]: This function provide only one way of sync (source ->destination)

File: main.cpp [hunk 6]
```diff
@@ -15,29 +18,70 @@ namespace fs = std::filesystem;
...
 
+    //Check recursively if the source and destination folders have the same number of subfolders
+    //Check modification time of all subfolder and compare it with the destination folder and its subfolders
+    for (auto &entry: fs::recursive_directory_iterator(source)) {
+        if (entry.is_directory()) {
+            auto path = entry.path();
+            auto relativePath = fs::relative(path, source);
+            auto destinationPath = destination / relativePath;
+            if (!fs::exists(destinationPath)) {
+                fs::create_directory(destinationPath);
```
Comment (mkijew) [inline main.cpp]: how about copying files ?

**Comment by GitBartlomiej** [comment]
All comments resolved. Ready to merge!

**Pull request merged**